src/tables.rs

```diff
@@ -199,3 +199,321 @@ pub fn bidi_l(c: char) -> bool {
         _ => false,
     }
 }
+
+/// The table of all Unicode combining characters, meaning all characters under
+/// the Unicode "Mark" (M) category.
+pub fn unicode_mark_category(c: char) -> bool {
```

Comment:
I'd probably pull in https://crates.io/crates/finl_unicode rather than inlining this.